mantidproject/mslice
# Interactive cut bug fix

Fixes two bugs in interactive (and normal) cuts.

1. Changing from interactive to normal mode did not restore the disabled `keep` / `make current` buttons.
2. If a cut is kept, the `CutPlotter` may still retain a reference to its `canvas` due to caching whereas the new plot data, due to the actions of the `FigureManager` has been sent to a new `canvas`, which is not updated. So the new plot is not shown.

**To test:**

Load a data file and make several cuts, keeping each one (two or three is enough). Then go back and `make current` one of the kept cuts and plot another cut - check that the plot updates as expected. Select another plot window to `make current`, then make a slice and click `Interactive Cut`. Check that the interactive cut is overplotting on the current plot window. Check that the `Keep` / `Make Current` buttons are hidden and a `Save` button is enabled. Move the interactive cut box around a check it works. Then click `Interactive Cut` again to disable it. Check that the `Keep` / `Make Current` buttons on the cut window are shown again.

Fixes #246.
